packages/foundry-sdk-generator/src/ontologyMetadata/ontologyMetadataResolver.ts

```diff
@@ -27,25 +33,30 @@ export class OntologyMetadataResolver {
     this.#authToken = authToken;
   }
 
-  private getRequestFunction<T>() {
-    return getApiRequestFunction<T>(
-      new UserTokenAuth({ userToken: this.#authToken }),
+  private getClientContext() {
+    return createClientContext(
+      {
+        metadata: {
+          userAgent: `foundry-typescript-osdk-generator/${process.env
+            .npm_package_version!}`,
+        },
+      },
       this.stackName,
+      () => this.#authToken,
       `foundry-typescript-osdk-generator/${process.env.npm_package_version!}`,
```

Comment:
Do we still need the user agent in both places?

Reply:
I started removing it in another PR. This felt like a complete piece of work and I was worried about making giant prs